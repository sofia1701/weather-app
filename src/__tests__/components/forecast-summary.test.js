import React from 'react';
import { render, cleanup } from "@testing-library/react";
import ForecastSummary from '../../components/forecast-summary';

afterEach(cleanup);

describe("ForecastSummary", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date="mockDate"
        temperature="mockTemperature"
        description="mockDescription"
        icon="mockIcon"
      />
    );
    expect(asFragment).toMatchSnapshot();
  })
  it("renders the correct props", () => {
    const { getByText } = render(
      <ForecastSummary
        date="mockDate"
        temperature="mockTemperature"
        description="mockDescription"
        icon="mockIcon"
      />
    );

    expect(getByText("mockDate")).toHaveClass("date");
    expect(getByText("mockTemperature")).toHaveClass("temperature");
    expect(getByText("mockDescription")).toHaveClass("description");
    expect(getByText("mockIcon")).toHaveClass("icon");
  })
});

