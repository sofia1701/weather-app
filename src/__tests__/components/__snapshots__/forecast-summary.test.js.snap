// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <div
      class="date"
      data-testid="date-id"
    >
      Monday 30th Apr
    </div>
    <div
      class="icon"
      data-testid="icon-id"
    >
      <i
        class="wi wi-day-sunny"
        name="owm"
      />
    </div>
    <div
      class="temperature"
      data-testid="temperature-id"
    >
      11Â°c
    </div>
    <div
      class="description"
      data-testid="description-id"
    >
      mockDescription
    </div>
    <button>
      More details
    </button>
  </div>
</DocumentFragment>
`;
